### PR TITLE
Added 'toggle' option to ContextMenu.

### PR DESCRIPTION
**Pull request by corwin-of-amber**
Following the discussion at https://github.com/os-js/osjs-panels/pull/26.

### CODE REVIEW

File: src/contextmenu.js [hunk 1]
```diff
@@ -182,9 +188,10 @@ export class ContextMenu {
         };
       },
       hide: () => props => {
-        this.callback = null;
+        if (isActive)
+          setTimeout(() => (isActive = false), 0);
         if (this.onclose) this.onclose();
-        this.onclose = null;
+        this.onclose = this.callback = null;
```
Comment (andersevenrud) [inline src/contextmenu.js]: Please use curly braces on all if-statements.

File: src/contextmenu.js [hunk 2]
```diff
@@ -150,8 +152,11 @@ export class ContextMenu {
         clampTimeout = timeout(() => clampSubMenu(this.core.$root, ev));
       },
       show: (options) => (props, actions) => {
-        let {menu, position} = options;
-        if (position instanceof Event) {
+        let {menu, position, toggle} = options;
+        if (toggle && isActive) {
+          return actions.hide();
```
Comment (andersevenrud) [inline src/contextmenu.js]: Place the `return;` on its own line.

File: src/contextmenu.js (hunk 1 above)
Comment (andersevenrud) [reply src/contextmenu.js]: Also, don't chain value assignments.

File: src/contextmenu.js (hunk 2 above)
Comment (corwin-of-amber) [reply src/contextmenu.js]: Uhm I thought the action has to return something. Initially it was `actions.hide(); return;` but then the return value from `hide` is lost and `show` returns `undefined`. It worked somehow, so perhaps it was ok?

Comment (corwin-of-amber) [reply src/contextmenu.js]: Also I noticed that sometimes you use `actions` (the argument) and sometimes `this.actions` (the member). Which is preferable?

Comment (andersevenrud) [reply src/contextmenu.js]: Yeah. In hindsight it's totally fine to just leave it there.

> Also I noticed that sometimes you use actions (the argument) and sometimes this.actions (the member). Which is preferable?

The former (argument) is probably a more natural way of doing this.

Comment (corwin-of-amber) [reply src/contextmenu.js]: Ok, I made all the actions use the argument and fixed the styling issues.

**Comment by corwin-of-amber** [comment]
I remembered to run `eslint` on it this time 🤓

**Comment by andersevenrud** [comment]
I see that I have to update the eslint config a little bit :angel:

File: src/contextmenu.js [hunk 3]
```diff
@@ -150,8 +152,11 @@ export class ContextMenu {
...
+        }
+        else if (position instanceof Event) {
```
Comment (andersevenrud) [inline src/contextmenu.js]: Else statements must be placed on the same line as the curly brace: `} else ...`

Comment (corwin-of-amber) [reply src/contextmenu.js]: Huh.

Comment (andersevenrud) [reply src/contextmenu.js]: As in

```
        if (toggle && isActive) {
          return actions.hide();
        } else if (position instanceof Event) {
```

**Comment by corwin-of-amber** [comment]
Am I supposed to indicate to Github somehow after I've done a requested change? I thought resolving the conversation does it. But it still says "Changes requested."

**Comment by andersevenrud** [comment]
I think it just bugged. On my end it can't find some reference to a commit. And there's no UI button or anything that lets me say OK or anything :man_facepalming: 

Which reminds me, you should probably make pull requests on custom branches, ex `feature/contextmenu-toggle`.

I might change some minor stuff or commit messages and the likes, so your master would not be up to sync with upstream properly in those cases.

**Comment by corwin-of-amber** [comment]
> your master would not be up to sync with upstream properly in those cases.

Ah yes, good thinking 👍

**Comment by corwin-of-amber** [comment]
Yayyyyy

**Comment by andersevenrud** [comment]
Thanks! :ok_hand:

**Comment by andersevenrud** [comment]
FYI: I updated the `@osjs/eslint-config` rules so that it triggers on all of the code style changes I've been requesting :)

**Comment by andersevenrud** [comment]
I just published a new version of `@osjs/gui` with these changes.

**Pull request merged**